packages/web/src/lib/web-mode.ts

```diff
@@ -184,62 +185,49 @@ export class WebMode {
-	 * @param txns array of multisignature transaction with at least one signature
+	 * @param txn Multisignature Encoded Transaction  
 	 * @param signers a subset of addresses to sign the transaction
-	 * return an object containing a blob attribute encoded in base64
+	 * return an object containing a blob key encoded in base64
 	 */
 	async appendSignMultisigTransaction(
-		txns: WalletTransaction[],
-		signers: string[]
+		txn: EncodedSignedTransaction,
+		signers?: string[]
 	): Promise<JsonPayload> {
 		try {
-			const result: JsonPayload = {};
-			for (let i = 0; i < txns.length; ++i) {
-				const txn = txns[i];
-				const partialTxn = algosdk.decodeObj(
-					this.algoSigner.encoding.base64ToMsgpack(txn.txn)
-				) as EncodedSignedTransaction;
-				if (partialTxn.txn === undefined || partialTxn.msig === undefined) {
-					throw new Error(
-						"Input transaction must be multisigature transaction signed with at least 1 signature"
-					);
-				}
-				const txnToBeSign = algosdk.Transaction.from_obj_for_encoding(partialTxn.txn);
-				const txnToBeSign_Uint8Array = algosdk.encodeObj(txnToBeSign.get_obj_for_encoding());
-				const txnToBeSign_Base64 =
-					this.algoSigner.encoding.msgpackToBase64(txnToBeSign_Uint8Array);
-
-				const mparams = partialTxn.msig as algosdk.EncodedMultisig;
-				const addrs = mparams.subsig.map((signData) => {
-					return algosdk.encodeAddress(signData.pk);
-				});
+			const encodedTxn = this.algoSigner.encoding.msgpackToBase64(encodeObj(txn.txn))
+			const mparams = txn.msig as algosdk.EncodedMultisig;
+			const version = mparams.v;
+			const threshold = mparams.thr;
+			const addr = mparams.subsig.map((signData) => {
+				return algosdk.encodeAddress(signData.pk)
+			});
+
+			const multisigParams = {
+				version: version,
+				threshold: threshold,
+				addrs: addr,
+			};
 
-				const multisigParams = {
-					version: mparams.v,
-					threshold: mparams.thr,
-					addrs: addrs,
-				};
-
-				const signedTxn = await this.signTransaction([
-					{
-						txn: txnToBeSign_Base64,
-						msig: multisigParams,
-						signers: signers,
-					},
-				]);
-
-				const signedTxnJson = signedTxn[0] as JsonPayload;
-				const blob = signedTxnJson.blob as string;
-
-				const blob1 = this.algoSigner.encoding.base64ToMsgpack(txn.txn);
-				const blob2 = this.algoSigner.encoding.base64ToMsgpack(blob);
-				const combineBlob = algosdk.mergeMultisigTransactions([blob1, blob2]);
-				const outputBase64 = this.algoSigner.encoding.msgpackToBase64(combineBlob);
-				result[i] = {
-					blob: outputBase64,
-				};
+			const signedTxn = await this.signTransaction([
+				{
+					txn: encodedTxn,
+					msig: multisigParams,
+					signers: signers,
+				},
+			]);
+
+			const signedTxnJson = signedTxn[0] as JsonPayload;
+			let combineBlob = this.algoSigner.encoding.base64ToMsgpack(signedTxnJson.blob as string)
+			// multiple signatures
+			if (txn.msig?.subsig.findIndex((item) => item.s?.length) !== -1) {
```

Comment:
From here, I think you can throw error like `Can not sign due to signer is not in the msig address`

Reply:
why this error? 
this code is to combine the blob (if the transaction provided is partially signed not completely)
and even if it is signed wholly, it throws error by itself

Reply:
Yes it should throw error but you can define the content of the error you want to throw so that user would know better what is the error instead of some throwing error can cause user to confuse

Reply:
I don't think this is the error which is appropriate here, because we are not really checking for the addr here, and also there can be the case of rekeyed account